client/components/data/query-all-jetpack-plugins/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { fetchAllPlugins } from 'calypso/state/plugins/installed/actions';
+import { isRequestingForAllSites } from 'calypso/state/plugins/installed/selectors';
+import type { CalypsoDispatch } from 'calypso/state/types';
+import type { AppState } from 'calypso/types';
+
+const request = () => ( dispatch: CalypsoDispatch, getState: AppState ) => {
+	if ( ! isRequestingForAllSites( getState() ) ) {
+		dispatch( fetchAllPlugins() );
+	}
+};
+
+export default function QueryAllJetpackPlugins() {
```

Comment:
The query component's name is a bit ambiguous. Am I requesting all plugins of this site? Or am I requesting the plugins for all sites? Is this about installed plugins or WP.org plugins available to install? IMHO this should be better reflected in the query component name.

Reply:
Thanks! What do you think about `QueryAllMyJetpackSitesPlugins`?

Reply:
Sounds good to me! We could ditch `My` though.

Reply:
I agree! Renamed in 8607109ef88edccbc285de74eee3b39809403ea9